src/parser/ParsedQuery.cpp

```diff
@@ -277,6 +278,10 @@ void ParsedQuery::GraphPattern::addLanguageFilter(const Variable& variable,
               QLEVER_INTERNAL_PREFIX_IRI_WITHOUT_CLOSING_BRACKET)) {
         matchingTriples.push_back(&triple);
       }
+      if (triple.s_ == variable || triple.o_ == variable ||
```

Comment:
And while you are at it:
At least add a TODO that there can be many more cases handled here, in particular
(Always assume the filter `FILTER (lang(?y) == "en")`
1. `?x <a>|<b> ?y`
2. `{?x <a> ?y } UNION { ?x <b> ?y}`
3. .....